### ISSUE TITLE
Name change for component library

### ISSUE THREAD

**Issue by KateEvans** [hmrc/component-library-template#14]
We'd like to change the name of the component library to make it more understandable and consistent with other resources. 

The name we're suggesting is 'HMRC elements'. This will be particularly beneficial to new starters who come on board and need to learn the various libraries and pattern repositories.

This change is suggested by the heads of design.

**Comment by feedmypixel** [comment]
@rpowis

**Comment by aluncoppack** [comment]
How about 'HMRC living style guide'? I saw this on https://medium.com/eightshapes-llc/team-models-for-scaling-a-design-system-2cf9d03be6a0#.3shp1dxfj and for me, it made it immediately apparent what this thing is and who should be consuming and contributing to it.

'[Elements](http://www.adobe.com/uk/products/photoshop-elements.html)' sounds like cheap photoshop, but maybe that's just me :)

**Comment by feedmypixel** [comment]
Just had a meeting with designer @NeilMcW. An interesting point was to align the content and naming conventions used on the component library to GOV.UK elements http://govuk-elements.herokuapp.com/ so design and frontend use GOV wide terminology for "elements" or "components".
With approaching it in this way 'HMRC elements' would align us with 'GOV.UK elements'

**Comment by aluncoppack** [comment]
An interesting philosophical question - when is something **adhering to convention** and when is something **doing it because we've always done it that way**?

**Comment by feedmypixel** [comment]
That is an interesting question especially with all the talk around component libraries.
 I suppose rather than adhering to convention it is for clarity of use and forming a consistent language for design and frontend to use, cross GOV. GOV.UK says apples but we call them pears. This is applicable to the finer details of the library as well as the name of the library. 
All great food for thought and discussion.

**Comment by feedmypixel** [comment]
A question that also came up is, are "elements" the same as "components"? Or are they two different things.

**Comment by rpowis** [comment]
> This will be particularly beneficial to new starters who come on board and need to learn the various libraries and pattern repositories.

@KateEvans I disagree. I think new starters would find greater benefit in clearer "Getting started" documentation than from library naming conventions.

> How about 'HMRC living style guide'? 

@aluncoppack I feel "Style guide" implies "_static governance_" whereas we want this to be an example of "_dynamic implementation_". Think "Tabs, not spaces!!" vs

```
  indentaion...
        who cares?

It'll look like this
```

...also, "living"? We need to explicitly state that it's "not dead"?

**Comment by rpowis** [comment]
> when is something **adhering to convention** and when is something **doing it because we've always done it that way**?

@aluncoppack It's the same, no?

**Comment by feedmypixel** [comment]
@rpowis an interesting point @NeilMcW mentioned is the use case for designers when creating and researching work. That use case was to have a look over http://govuk-elements.herokuapp.com/ and see what the guidelines are and then create something and work with what we have in assets-frontend. In that context you can see how it becomes a little confusing with naming of the "library" and the naming of "components" or "elements".

> An interesting point was to align the content and naming conventions used on the component library to GOV.UK elements http://govuk-elements.herokuapp.com/

Maybe a wider discussion needs to happen about this and about the task of aligning say `.input--normal` http://hmrc.github.io/assets-frontend/section-text-input.html with `.form-control` http://govuk-elements.herokuapp.com/form-elements/#telephone-number. Neil really pointed out how important this is for communication across design and frontend for what is being talked about/worked on. This potentially also reflects on the "component library"
